src/rnn/mikolov_rnn.cc

```diff
@@ -177,29 +177,33 @@ Status MikolovModelReader::parse() {
   auto maxBlock = std::max<u64>({hdr.layerSize * hdr.vocabSize, hdr.maxentSize,
                                  hdr.layerSize * hdr.layerSize});
   // 3 comes from rounding to next value + sizeof(float)
-  auto pageSize = 1ULL << (static_cast<u32>(std::log2(maxBlock)) + 3);
+  auto pageSize = static_cast<size_t>(1) << (static_cast<u32>(std::log2(maxBlock)) + 3);
   data_->memmgr.initialize(pageSize);
   data_->alloc = data_->memmgr.value().core();
   auto& alloc = data_->alloc;
+
+  const auto embedding_size = static_cast<size_t>(hdr.layerSize) * static_cast<size_t>(hdr.vocabSize);
+  const auto matrix_size = static_cast<size_t>(hdr.layerSize) * static_cast<size_t>(hdr.layerSize);
```

Comment:
I had second thought about whether `hdr.layerSize * hdr.layerSize` is mistake or not...

Reply:
It's a square matrix layerSize times layerSize, so that's correct